src/qtism/data/storage/xml/filesystem/FlysystemV2Filesystem.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace qtism\data\storage\xml\filesystem;
+
+use Exception;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemException as FlysystemException;
+use League\Flysystem\Local\FallbackMimeTypeDetector;
+use League\Flysystem\Local\LocalFilesystemAdapter;
+use League\MimeTypeDetection\ExtensionMimeTypeDetector;
+
+class FlysystemV2Filesystem implements FilesystemInterface
+{
+    protected Filesystem $filesystem;
+
+    public function __construct(Filesystem $filesystem)
+    {
+        $this->filesystem = $filesystem;
+    }
+
+    public static function local(string $path = '/'): self
+    {
+        return new self(
+            new Filesystem(
+                new LocalFilesystemAdapter(
+                    $path,
+                    null,
+                    LOCK_EX,
+                    LocalFilesystemAdapter::DISALLOW_LINKS,
+                    new FallbackMimeTypeDetector(new ExtensionMimeTypeDetector()), // Serializable
```

Comment:
Two issues I failed to spot here that break the 7.4 compatibility, which we have to care about for the time being.
1. A trailing comma in the arguments' list
2. [FallbackMimeTypeDetector was added in flysystem v3 only](https://github.com/thephpleague/flysystem/commit/e8515341994dc2feb13a5723316ae521841eda0f#diff-9a576af08e4ecea9ed448f74e44c164f445a85ec447b419e8dd385d7b480685d)